langchain/src/chat_models/openai.ts

```diff
@@ -470,7 +492,25 @@ export class ChatOpenAI<
       const generations = Object.entries(finalChunks)
         .sort(([aKey], [bKey]) => parseInt(aKey, 10) - parseInt(bKey, 10))
         .map(([_, value]) => value);
-      return { generations };
+
+      const { functions, function_call } = this.invocationParams(options);
+
+      // OpenAI does not support token usage report under stream mode,
+      // fallback to estimation.
+
+      const promptTokenUsage = await this.getNumTokensFromPrompt(
+        messages,
+        functions,
+        function_call
+      );
+      const completionTokenUsage = await this.getNumTokensFromGenerations(
+        generations
+      );
+
+      tokenUsage.promptTokens = promptTokenUsage;
```

Comment:
Since it's estimated, could we change these properties to e.g. `estimatedPromptTokens`?

I suspect we may see some updates at developer day next week